perf-event/src/events/msr.rs

```diff
@@ -0,0 +1,68 @@
+use crate::events::Event;
+use c_enum::c_enum;
+use perf_event_open_sys::bindings;
+use std::io;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+// 0 will never be the PMU value for msr.
+// We use it as a flag value to indicate that this has not been initialized.
+static MSR_TYPE: AtomicU32 = AtomicU32::new(0);
+
+
+c_enum! {
+    /// MSR events on all x86 CPUs
+    #[derive(Clone, Copy, Eq, PartialEq, Hash)]
+    pub enum MSRConfig: u64 {
+        /// Time Stamp Counter
+        /// TSC increases at the CPU base frequency.
+        TSC = 0x0,
+        /// Actual Performance Frequency Clock Count
+        APERF = 0x1,
+        /// Maximum Performance Frequency Clock Count
+        /// (APERF / MPERF) * CPU_BASE_FREQUENCY is the running CPU frequency
+        MPERF = 0x2,
```

Comment:
The docs here should also mention which x86 MSR counter they correspond to. See [this table][0] in the kernel source. Ideally they would also link to the section in the Intel SDM documenting these MSRs but that isn't required.

[0]: https://sourcegraph.com/github.com/torvalds/linux@9db898594c541444e19b2d20fb8a06262cf8fcd9/-/blob/arch/x86/events/msr.c?L155